### PR TITLE
update counter example to use react-redux v0.5.1

### PR DESCRIPTION
**Pull request by hartzis**
Update Counter example with most recent react-redux.

Based on https://github.com/gaearon/react-redux/pull/29. 

Remove `CounterApp` component and instead connect to the `Counter` component itself.

I was also debating using the example with Object.assign, but didn't want to mess with all the different ways to bring that in. Destructuring off actions seems to work alright.

Let me know what you think.

### CODE REVIEW

**Comment by hartzis** [comment]
I missed updating the tests, working on them now.

**Comment by hartzis** [comment]
@gaearon Just saw you updated react-redux to [v0.6.0](https://github.com/gaearon/react-redux/releases/tag/v0.6.0). I'll update this request in the next couple of days.

File: examples/counter/components/Counter.js [hunk 1]
```diff
@@ -1,9 +1,12 @@
 import React, { Component, PropTypes } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import * as CounterActions from '../actions/CounterActions';
 
-class Counter extends Component {
+export class CounterComponent extends Component {
```
Comment (gaearon) [inline examples/counter/components/Counter.js]: Let's keep calling it `Counter`?

File: examples/counter/containers/CounterApp.js [hunk 2]
```diff
@@ -1,23 +0,0 @@
-import React, { Component } from 'react';
```
Comment (gaearon) [inline examples/counter/containers/CounterApp.js]: I'd do it differently. Let's encourage separation of “smart” and “dumb” components. See https://github.com/gaearon/react-redux#dumb-component-is-unaware-of-redux and https://github.com/gaearon/react-redux#smart-component-is-connect-ed-to-redux.

**Comment by hartzis** [comment]
@gaearon Definitely agree with the "smart" and "dumb" component separation. I think this now appears to match your react-redux readme example. Let me know what you think.

**Comment by gaearon** [comment]
Any specific reason you're putting them in `actions` instead of just spreading, like before?

**Comment by hartzis** [comment]
It seemed more explicit assigning actions to an actions prop, [example](https://github.com/gaearon/react-redux#inject-todos-and-all-action-creators-addtodo-completetodo--as-actions). 

I also didn't want to require an 'Object.Assign' library, but now I realize we could do this instead:

```
function mapDispatchToProps(dispatch) {
  return {
    ...bindActionCreators(CounterActions, dispatch)
  }
}
```
vs
https://github.com/gaearon/react-redux#inject-todos-and-all-todoactioncreators-and-counteractioncreators-directly-as-props
```
function mapDispatchToProps(dispatch) {
  return bindActionCreators(Object.assign({}, todoActionCreators, counterActionCreators), dispatch);
}
```

I do like the spread operator for `mapDispatchToProps`.

**Comment by gaearon** [comment]
This:

```js
function mapDispatchToProps(dispatch) {
  return {
    ...bindActionCreators(CounterActions, dispatch)
  }
}
```

is exactly equivalent to this:

```js
function mapDispatchToProps(dispatch) {
  return bindActionCreators(CounterActions, dispatch);
}
```

There is no need for spread here.

**Comment by hartzis** [comment]
Thank you sir, nice catch.

Sounds good. Fixed spreading actions into `Counter`.

**Comment by hartzis** [comment]
@gaearon Would you like me to resubmit/squash this pull request to a single commit? I realize only `CounterApp` and `package.json` should have been updated.

**Comment by gaearon** [comment]
Yes, please!

**Comment by gaearon** [comment]
Did you mean to open a new PR?

**Comment by hartzis** [comment]
I did, https://github.com/gaearon/redux/pull/453. I thought it would be easier/cleaner.

**Comment by gaearon** [comment]
Thanks!

**Pull request closed without merging**